File: CobongMemo/app/src/main/java/cobong/jeongwoojin/cobongmemo/cobongmemo/view/memo/MemoViewModel.java
package cobong.jeongwoojin.cobongmemo.cobongmemo.view.memo;

import androidx.lifecycle.ViewModel;
import cobong.jeongwoojin.cobongmemo.cobongmemo.model.MemoListItem;

public class MemoViewModel extends ViewModel {

    MemoNavigator navigator;

    public void sendMemo(MemoListItem item) {
        navigator.sendMemo(item);
    }

    public void onDeleteMemoClick(MemoListItem memo) {
        navigator.deleteMemo(memo);
    }

    public void onEditMemoClick(MemoListItem memo) {
        navigator.editMemo(memo);
    }

    public MemoNavigator getNavigator() {
        return navigator;
    }

    public void setNavigator(MemoNavigator navigator) {
        this.navigator = navigator;
    }
}

File: CobongMemo/app/src/main/java/cobong/jeongwoojin/cobongmemo/cobongmemo/view/MemoViewModel.kt
package cobong.jeongwoojin.cobongmemo.cobongmemo.view

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import cobong.jeongwoojin.cobongmemo.cobongmemo.model.MemoItem
import cobong.jeongwoojin.cobongmemo.cobongmemo.model.MemoRepository
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class MemoViewModel(application: Application) : AndroidViewModel(application) {

    var navigator: MemoNavigator? = null
    private var repository: MemoRepository

    val allMemosByRoom: LiveData<List<MemoItem>>

    init {
        repository = MemoRepository.getInstance(application)
        allMemosByRoom = repository.memos
    }

    fun deleteByRoom(memo: MemoItem) = viewModelScope.launch(Dispatchers.IO) {
        repository.deleteByRoom(memo)
    }

    override fun onCleared() {
        super.onCleared()
    }


    fun sendMemo(item: MemoItem) {
        navigator!!.sendMemo(item)
    }

    fun onDeleteMemoClick(memo: MemoItem) {
        navigator!!.deleteMemo(memo)
    }

    fun onEditMemoClick(memo: MemoItem) {
        navigator!!.editMemo(memo)
    }
}
